src/main/java/org/jabref/gui/preferences/PreviewTabView.java

```diff
@@ -0,0 +1,250 @@
+                  .root(this)
+                  .load();
+    }
+
+    public void initialize() {
+        viewModel = new PreviewTabViewModel(dialogService, preferences, taskExecutor);
+
+        availableListView.itemsProperty().bind(viewModel.availableListProperty());
+        viewModel.selectedAvailableItemsProperty().bind(new ReadOnlyListWrapper(availableListView.getSelectionModel().getSelectedItems()));
+        new ViewModelListCellFactory<PreviewLayout>().withText(PreviewLayout::getName).install(availableListView);
+
+        chosenListView.itemsProperty().bind(viewModel.chosenListProperty());
+        viewModel.selectedChosenItemsProperty().bind(new ReadOnlyListWrapper(chosenListView.getSelectionModel().getSelectedItems()));
+        new ViewModelListCellFactory<PreviewLayout>().withText(PreviewLayout::getName).install(chosenListView);
+
+        toRightButton.disableProperty().bind(availableListView.selectionModelProperty().getValue().selectedItemProperty().isNull());
+
+        BooleanBinding nothingSelectedFromChosen = chosenListView.selectionModelProperty().getValue().selectedItemProperty().isNull();
+        toLeftButton.disableProperty().bind(nothingSelectedFromChosen);
+        sortUpButton.disableProperty().bind(nothingSelectedFromChosen);
+        sortDownButton.disableProperty().bind(nothingSelectedFromChosen);
+        testButton.disableProperty().bind(nothingSelectedFromChosen);
+        defaultButton.disableProperty().bind(nothingSelectedFromChosen);
+
+        previewTextArea.setParagraphGraphicFactory(LineNumberFactory.get(previewTextArea));
+        previewTextArea.textProperty().addListener((obs, oldText, newText) -> {
+            previewTextArea.setStyleSpans(0, computeHighlighting(newText));
+        });
+
+        BindingsHelper.bindBidirectional(previewTextArea.textProperty(), viewModel.previewTextProperty(),
+                text -> previewTextArea.replaceText(viewModel.previewTextProperty().getValue()),
+                text -> viewModel.previewTextProperty().setValue(previewTextArea.getText()));
+
+        ActionFactory factory = new ActionFactory(Globals.getKeyPrefs());
+        ContextMenu contextMenu = new ContextMenu();
+        contextMenu.getItems().addAll(
+                factory.createMenuItem(StandardActions.CUT, new PreviewTabView.EditAction(StandardActions.CUT)),
+                factory.createMenuItem(StandardActions.COPY, new PreviewTabView.EditAction(StandardActions.COPY)),
+                factory.createMenuItem(StandardActions.PASTE, new PreviewTabView.EditAction(StandardActions.PASTE)),
+                factory.createMenuItem(StandardActions.SELECT_ALL, new PreviewTabView.EditAction(StandardActions.SELECT_ALL))
+        );
+
+        contextMenu.getStyleClass().add("context-menu");
+        previewTextArea.setContextMenu(contextMenu);
+    }
+
+    /**
+     * This was originally created by Carlos Martins (github: @cemarins )
+     * Can this be used here? (BSD-2clause License)
```

Comment:
Yep. BSD2-clause aka FreeBSD is compatible

Reply:
So I can cut the comment? Where do I give credit to him?

Reply:
I would suggest extending the javadoc comment, e.g. author copyright, and link to the code 
Not sure if you need the whole Disclaimer or licence text.
https://opensource.org/licenses/BSD-2-Clause
That should be enough. We already link to the library in our external-libraries.txt file